test/utils/datetime.test.js

```diff
@@ -332,4 +332,24 @@ describe('datetime', () => {
         .to.throw('time and deviceTime must be a ISO-formatted String timestamp');
     });
   });
+
+  describe('midDayForDate', () => {
```

Comment:
Please add more test cases here to prove that this behaves as we desire for the edge cases of the day that we switch to DST and from DST. The goal is that in the case of the foreshortened day (23 hrs), you still see midnight on the left edge (-432e5 milliseconds from result, not 12 "clock" hours). Ditto for the lengthened day (25 hrs).

Also as a tip: it's generally best to use a timezone in tests that none of us is in. This prevents tests that are passing only due to the fact that you're in the timezone you're using in the tests. They then end up failing on Travis, causing much 😭. Obviously that didn't happen here, but just as a tip. [I often use Mountain or Central time (or CEST) to guard against this...]